File: generated-code/python/codecraft/FileReadWrite/color.py
class Color:
    """RGBA Color"""

    def __init__(self, r, g, b, a):
        self.r = r
        """Red component"""
        self.g = g
        """Green component"""
        self.b = b
        """Blue component"""
        self.a = a
        """Alpha (opacity) component"""

    @staticmethod
    def read_from(stream):
        """Read Color from input stream
        """
        r = stream.read_float()
        g = stream.read_float()
        b = stream.read_float()
        a = stream.read_float()
        return Color(r, g, b, a)
    
    def write_to(self, stream):
        """Write Color to output stream
        """
        stream.write_float(self.r)
        stream.write_float(self.g)
        stream.write_float(self.b)
        stream.write_float(self.a)
    
    def __repr__(self):
        return "Color(" + \
            repr(self.r) + \
            ", " + \
            repr(self.g) + \
            ", " + \
            repr(self.b) + \
            ", " + \
            repr(self.a) + \
            ")"

File: generated-code/ruby/codecraft/FileReadWrite/color.rb
# RGBA Color
class Color
    # Red component
    attr_accessor :r
    # Green component
    attr_accessor :g
    # Blue component
    attr_accessor :b
    # Alpha (opacity) component
    attr_accessor :a

    def initialize(r, g, b, a)
        @r = r
        @g = g
        @b = b
        @a = a
    end

    # Read Color from input stream
    def self.read_from(stream)
        r = stream.read_float()
        g = stream.read_float()
        b = stream.read_float()
        a = stream.read_float()
        Color.new(r, g, b, a)
    end

    # Write Color to output stream
    def write_to(stream)
        stream.write_float(@r)
        stream.write_float(@g)
        stream.write_float(@b)
        stream.write_float(@a)
    end

    def to_s
        string_result = "Color { "
        string_result += "r: "
        string_result += @r.to_s
        string_result += ", "
        string_result += "g: "
        string_result += @g.to_s
        string_result += ", "
        string_result += "b: "
        string_result += @b.to_s
        string_result += ", "
        string_result += "a: "
        string_result += @a.to_s
        string_result += " }"
        string_result
    end

    def to_str
        to_s
    end
end